Makefile

```diff
@@ -84,19 +87,19 @@ ${BUILD_DIR}/%.o: ${APP_BASE}/source/%.c
 	@echo "[application] Building: $<"
 	@${CC} ${CFLAGS} -c $< -o $@
 
-${APP_LIB}: ${APP_OBJ} version
+${APP_LIB}: ${APP_VER} ${APP_OBJ}
 	@echo "[application] Linking app library"
-	@${AR} qc $@ $^
+	@${AR} rc $@ $?
 
 dir:
 	@mkdir -p ${BUILD_DIR}
 
-version: dir
-	@./version.py CS704 ${BUILD_DIR}/cs704_version.h
+${APP_VER}: dir
```

Comment:
In hindsight, this may no longer update the version number for a non clean build because of dependencies; but it seemed to still run fine when I tried it.